pkg/logging/kinesis/delete.go

```diff
@@ -0,0 +1,47 @@
+package kinesis
+
+import (
+	"io"
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/v2/fastly"
+)
+
+// DeleteCommand calls the Fastly API to delete an Amazon Kinesis logging endpoint.
+type DeleteCommand struct {
+	common.Base
+	manifest manifest.Data
+	Input    fastly.DeleteKinesisInput
+}
+
+// NewDeleteCommand returns a usable command registered under the parent.
+func NewDeleteCommand(parent common.Registerer, globals *config.Data) *DeleteCommand {
+	var c DeleteCommand
+	c.Globals = globals
+	c.manifest.File.Read(manifest.Filename)
+	c.CmdClause = parent.Command("delete", "Delete a Kinesis logging endpoint on a Fastly service version").Alias("remove")
+	c.CmdClause.Flag("service-id", "Service ID").Required().Short('s').StringVar(&c.manifest.Flag.ServiceID)
```

Comment:
@kellymclaughlin so interestingly I'm not sure if service-id needs to be 'required' in the sense that later on in `Exec` we have...

```go
serviceID, source := c.manifest.ServiceID()
if source == manifest.SourceUndefined {
	return errors.ErrNoServiceID
}
```

If we inspect the `ServiceID()` method it's checking first the flag and _then_ the manifest file for a service id before erroring if both are missing...

https://github.com/fastly/cli/blob/c0f337321376bb029c1ffee8b8c1181289dff695/pkg/compute/manifest/manifest.go#L52-L63

Reply:
Oh interesting. Ok, I'll remove the `Required` and move the `service-id` option to the optional section then.

Reply:
Done